### PR TITLE
add driver for keithley6500

### PR DESCRIPTION
**Pull request by peendebak**
@lucblom @astafan8 @WilliamHPNielsen

### CODE REVIEW

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                           get_parser=float,
+                           get_cmd='ROUT:SCAN:INT?',
+                           set_cmd='ROUT:SCAN:INT {}',
+                           unit='s', vals=Numbers(min_value=0.001, max_value=999999.999))
+
+        self.add_parameter('amplitude',
+                           get_cmd=self._read_next_value,
+                           unit='arb.unit')
+
+        self.add_function('reset', call_cmd='*RST')
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]: please don't use function, but a bound method instead. we are deprecating functions.

**Comment by peendebak** [comment]
@Christian-Volk I addressed some of the comments, can you look at the remaining comments?

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+from qcodes.utils.validators import Bool, Enum, Ints, MultiType, Numbers
+
+
+def _parse_output_string(string_value):
+    """ Parses and cleans string output of the multimeter. Removes the surrounding
+        whitespace, newline characters and quotes from the parsed data. Some results
+        are converted for readablitity (e.g. mov changes to moving).
+
+    Args:
+        string_value (str): The data returned from the mulitmeter reading commands.
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]:
```suggestion
        string_value: The data returned from the multimeter reading commands.
```

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
+from functools import partial
+
+from qcodes import VisaInstrument
+from qcodes.utils.validators import Bool, Enum, Ints, MultiType, Numbers
+
+
+def _parse_output_string(string_value):
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]:
```suggestion
def _parse_output_string(string_value: str):
```

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    """ Parses and cleans string output of the multimeter. Removes the surrounding
+        whitespace, newline characters and quotes from the parsed data. Some results
+        are converted for readablitity (e.g. mov changes to moving).
+
+    Args:
+        string_value (str): The data returned from the mulitmeter reading commands.
+
+    Returns:
+        String: The cleanup output data.
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]:
```suggestion
        The cleaned-up output of the multimeter.
```

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    if (s[0] == s[-1]) and s.startswith(("'", '"')):
+        s = s[1:-1]
+
+    conversions = {'mov': 'moving', 'rep': 'repeat'}
+    if s in conversions.keys():
+        s = conversions[s]
+    return s
+
+
+def parse_output_bool(numeric_value):
```
[the remark this answers is not in the document]
Comment (astafan8) [reply qcodes/instrument_drivers/tektronix/Keithley_6500.py]: please make it private :)

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]:
```suggestion
def parse_output_bool(numeric_value: Union[int, float]):
```

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    """ Parses and converts the value to boolean type. True is 1.
+
+    Args:
+        numeric_value (int, float): The numberical value to convert.
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]:
```suggestion
        numeric_value: The numerical value to convert.
```

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+    Returns:
+        Bool: The cleanup output data.
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]:
```suggestion
        The boolean representation of the numeric value.
```

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+    """
+    return int(numeric_value) == 1
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]: why didn't you simply use `bool(numeric_value)`? Doesn't it work as you expect?

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+                           unit='arb.unit')
+
+        self.add_function('reset', call_cmd='*RST')
+
+        if reset:
+            self.reset()
+        self.write('FORM:DATA ASCII')
+        self.connect_message()
+
+    def _read_next_value(self):
```
[the remark this answers is not in the document]
Comment (astafan8) [reply qcodes/instrument_drivers/tektronix/Keithley_6500.py]: ok. it's fine like this, but next time use simple conveniences that QCoDeS provides :)

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+    if (s[0] == s[-1]) and s.startswith(("'", '"')):
+        s = s[1:-1]
+
+    conversions = {'mov': 'moving', 'rep': 'repeat'}
+    if s in conversions.keys():
+        s = conversions[s]
+    return s
+
+
+def _parse_output_bool(numeric_value: Union[int, float]):
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]: There is no need for this function, right? `bool` itself will work just as good and will save 10 lines of code, and will make the rest of the driver more readable.

**Pull request merged**